### PR TITLE
DOCS: Update Gamepad documentation

### PR DESCRIPTION
**Pull request by jfreire-unity**
## Documentation fixes
Updates Gamepad documentation based on Unity Package API Quality Standards by addressing Score Explanation issues:
- Fix related links for all, buttonEast, buttonNorth, buttonSouth, buttonWest, current
- Removed value tag for aButton, all, bButton, buttonEast, buttonNorth, buttonSouth, buttonWest, circleButton, crossButton, current, dpad, leftShoulder, leftStick, leftStickButton, leftTrigger, rightShoulder, rightStick, rightStickButton, rightTrigger, selectButton, squareButton, startButton, triangleButton, xButton, yButton.
- Fixed insufficient remarks length (threshold is 120) on MakeCurrent()
- Added related links (required for this API type) to Gamepad class
- Added  code samples (required for this API type) each for MakeCurrent(), PauseHaptics(), ResetHaptics(), ResumeHaptics(), SetMotorSpeeds(float, float)
  - ❌ Decided not to add examples to FinishSetup(),  OnAdded(), OnRemoved() since they are protected methods.
- Added remarks  for OnAdded(), OnRemoved(), PauseHaptics(), ResetHaptics(), ResumeHaptics()
- Expanded code samples (required for this API type) for Gamepad class.

## Reusability of code examples improvement

This PR follows the approach mentioned in https://docs.unity.com/internal/content-ops/en-us/documentation/content-model/code-snippets-examples.

Many packages such as the Localization and Entities packages follow this convention. I'm throwing the first rock for our package :D 

Also, see the [Slack thread](https://unity.slack.com/archives/C08100WBU92/p1733326529294579) for more insights in this approach, so that I could fix PVP problems.

I'm open for deeper improvements in follow-up PRs but this seems like a good way forward for now.
Let me know your thoughts. 

### Testing status & QA

Docs build on my machine.

### Overall Product Risks

None

- Complexity: Minimal
- Halo Effect: Minimal

### Comments to reviewers

Please let me know of any inconsistencies and improvements I should do.
Also, let me know if the approach to reuse code samples in the folder `DocCodeSamples` is ok.
### Checklist

Before review:

- [x] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [x] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

After merge:

- [ ] Create forward/backward port if needed. If you are blocked from creating a forward port now please add a task to ISX-1444.

### CODE REVIEW

File: Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs [hunk 1]
```diff
@@ -132,20 +131,18 @@ public struct GamepadState : IInputStateTypeInfo
...
         [FieldOffset(4)]
         public Vector2 leftStick;
 
         /// <summary>
-        /// Right stick position. Each axis from -1 to 1 with
-        /// 0 being center position.
+        /// Right stick position.
         /// </summary>
-        /// <value>Right stick position.</value>
+        /// <remarks>Each axis from -1 to 1 with 0 being center position.</remarks>
```
Comment (stefanunity) [inline Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs]: remarks for left and right stick axis are not the same

File: Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs [hunk 2]
```diff
@@ -402,106 +399,146 @@ namespace UnityEngine.InputSystem
...
+    ///
+    ///         // Check if the button control is being continuously actuated and read its value
+    ///         if (gamepad.rightTrigger.IsActuated())
+    ///         {
+    ///             Debug.Log("Right trigger value: " + gamepad.rightTrigger.ReadValue());
+    ///         }
+    ///
+    ///         // Read left stick value and perform some code based on the value
+    ///         Vector2 move = gamepad.leftStick.ReadValue();
+    ///         {
```
Comment (stefanunity) [inline Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs]: why { } if there is no control statement?

**Review by stefanunity — APPROVED** [review]

File: Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs (hunk 2 above)
Comment (jfreire-unity) [reply Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs]: It's just to scope out  the`move` variable, as seen in an example [here](https://docs.unity3d.com/Packages/com.unity.inputsystem@1.11/manual/Workflow-Direct.html)

File: Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs [hunk 3]
```diff
@@ -174,7 +171,7 @@ public struct GamepadState : IInputStateTypeInfo
         /// <summary>
         /// State format tag for GamepadState.
         /// </summary>
-        /// <value>Returns "GPAD".</value>
+        /// <remarks>Returns "GPAD".</remarks>
```
Comment (ekcoh) [inline Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs]: I would guess this violates length requirement, is a remark required here? Could it be "State format tag for GamepadState ("GPAD")."

File: Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs [hunk 4]
```diff
@@ -402,106 +399,146 @@ namespace UnityEngine.InputSystem
...
+    ///         if (gamepad.buttonSouth.wasPressedThisFrame)
+    ///         {
+    ///             gamepad.SetMotorSpeeds(0.2f, 1.0f);
+    ///
+    ///         } else if (gamepad.buttonSouth.wasReleasedThisFrame)
+    ///         {
+    ///             gamepad.ResetHaptics();
+    ///         }
+    ///     }
+    /// }
```
Comment (ekcoh) [inline Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs]: I am afraid example nested inside remarks will generate CI xmldcc error?

File: Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs [hunk 5]
```diff
@@ -402,106 +399,146 @@ namespace UnityEngine.InputSystem
...
         public ButtonControl buttonWest { get; protected set; }
 
         /// <summary>
         /// The top face button of the gamepad.
         /// </summary>
-        /// <value>Control representing the Y/Triangle face button.</value>
         /// <remarks>
-        /// On an Xbox controller, this is the Y button and on the PS4 controller, this is the
-        /// triangle button.
+        /// Control representing the Y/Triangle face button.
```
Comment (ekcoh) [inline Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs:432]: Side-note: I think this doc clearly illustrates the problem with current aliased enums. E.g. imagine if we supported 800 gamepads, how many variants of the mapping should be mentioned here? :)

File: Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs [hunk 6]
```diff
@@ -730,9 +788,19 @@ protected override void OnRemoved()
...
+        /// <inheritdoc cref="DualMotorRumble.PauseHaptics"/>
         /// </summary>
+        /// <remarks>
+        /// Resume with <see cref="ResumeHaptics"/>.
+        /// <inheritdoc cref="DualMotorRumble.PauseHaptics"/>
+        /// </remarks>
         /// <seealso cref="IDualMotorRumble"/>
+        /// <example>
+        /// <code>
+        /// <inheritdoc cref="SetMotorSpeeds"/>
```
Comment (ekcoh) [inline Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs]: Nice one

**Review by ekcoh — APPROVED** [review]

**Review by ekcoh — CHANGES_REQUESTED** [review]
Seems like there are more work to do unfortunately, see dry-run CI job:


PVP-150-1: UnityEngine.InputSystem.Gamepad: void OnAdded(): <inheritdoc> in block context (only allowed in top-level context)
PVP-150-1: UnityEngine.InputSystem.Gamepad: void OnRemoved(): <inheritdoc> in block context (only allowed in top-level context)
PVP-150-1: UnityEngine.InputSystem.Gamepad: void PauseHaptics(): <inheritdoc> in block context (only allowed in top-level context)
PVP-150-1: UnityEngine.InputSystem.Gamepad: void PauseHaptics(): tag inside <code>
PVP-150-1: UnityEngine.InputSystem.Gamepad: void ResumeHaptics(): <inheritdoc> in block context (only allowed in top-level context)
PVP-150-1: UnityEngine.InputSystem.Gamepad: void ResumeHaptics(): tag inside <code>
PVP-150-1: UnityEngine.InputSystem.Gamepad: void ResetHaptics(): <inheritdoc> in block context (only allowed in top-level context)
PVP-150-1: UnityEngine.InputSystem.Gamepad: void ResetHaptics(): tag inside <code>

File: Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs (hunk 4 above)
Comment (jfreire-unity) [reply Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs]: Ah didn't notice it, will fix

File: Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs (hunk 5 above)
Comment (jfreire-unity) [reply Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs:432]: Yeah, but I think this only used for convenience. The aliases could (or maybe should)  be removed in a 2.x

File: Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs (hunk 3 above)
Comment (jfreire-unity) [reply Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs]: Fixed

**Comment by jfreire-unity** [comment]
> Seems like there are more work to do unfortunately, see dry-run CI job:
> 
> PVP-150-1: UnityEngine.InputSystem.Gamepad: void OnAdded(): in block context (only allowed in top-level context) PVP-150-1: UnityEngine.InputSystem.Gamepad: void OnRemoved(): in block context (only allowed in top-level context) PVP-150-1: UnityEngine.InputSystem.Gamepad: void PauseHaptics(): in block context (only allowed in top-level context) PVP-150-1: UnityEngine.InputSystem.Gamepad: void PauseHaptics(): tag inside `` PVP-150-1: UnityEngine.InputSystem.Gamepad: void ResumeHaptics(): in block context (only allowed in top-level context) PVP-150-1: UnityEngine.InputSystem.Gamepad: void ResumeHaptics(): tag inside ` PVP-150-1: UnityEngine.InputSystem.Gamepad: void ResetHaptics(): in block context (only allowed in top-level context) PVP-150-1: UnityEngine.InputSystem.Gamepad: void ResetHaptics(): tag inside ` ``

Missed those, will fix them as soon as I understand them :D

File: Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs [hunk 7]
```diff
@@ -154,18 +151,18 @@ public struct GamepadState : IInputStateTypeInfo
         ////REVIEW: should left and right trigger get deadzones?
 
         /// <summary>
-        /// Position of the left trigger. Goes from 0 (not pressed) to 1 (fully pressed).
+        /// The current position of the left trigger on a gamepad.
         /// </summary>
-        /// <value>Position of left trigger.</value>
+        /// <remarks>The value's range goes from 0 to 1, where 0 represents the trigger not pressed at all, and 1 represents the trigger in its fully pressed position.</remarks>
```
Comment (ritamerkl) [inline Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs]: not pressed at all -> in it's neutral position, its -> it's

Comment (ritamerkl) [reply Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs]: likewise in other parts of the code.

File: Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs [hunk 8]
```diff
@@ -637,28 +663,29 @@ public ButtonControl this[GamepadButton button]
...
         /// <remarks>
+        /// Returns all currently connected gamepads.
+        ///
         /// Does not cause GC allocation.
         ///
         /// Do <em>not</em> hold on to the value returned by this getter but rather query it whenever
         /// you need it. Whenever the gamepad setup changes, the value returned by this getter
         /// is invalidated.
+        ///
+        /// Alternately, if you want a single gamepad, you can use <see cref="current"/> for example.
```
Comment (ritamerkl) [inline Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs]: "if you want a single gamepad" sounds a bit odd. maybe "for querying a single gamepad"?

File: Packages/com.unity.inputsystem/InputSystem/Devices/Haptics/DualMotorRumble.cs [hunk 9]
```diff
@@ -39,9 +39,13 @@ internal struct DualMotorRumble
             || !Mathf.Approximately(highFrequencyMotorSpeed, 0f);
 
         /// <summary>
-        /// Reset motor speeds to zero but retain current values for <see cref="lowFrequencyMotorSpeed"/>
-        /// and <see cref="highFrequencyMotorSpeed"/>.
+        /// Reset motor speeds to zero.
```
Comment (ritamerkl) [inline Packages/com.unity.inputsystem/InputSystem/Devices/Haptics/DualMotorRumble.cs]: Is this summary long enough to fit the requirements in count of characters?

**Review by ritamerkl — APPROVED** [review]
LGTM, nice code examples

File: Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs (hunk 7 above)
Comment (jfreire-unity) [reply Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs]: Fixed

File: Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs (hunk 8 above)
Comment (jfreire-unity) [reply Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs]: Thanks! Fixed

**Comment by jfreire-unity** [comment]
> Seems like there are more work to do unfortunately, see dry-run CI job:
> 
> PVP-150-1: UnityEngine.InputSystem.Gamepad: void OnAdded(): in block context (only allowed in top-level context) PVP-150-1: UnityEngine.InputSystem.Gamepad: void OnRemoved(): in block context (only allowed in top-level context) PVP-150-1: UnityEngine.InputSystem.Gamepad: void PauseHaptics(): in block context (only allowed in top-level context) PVP-150-1: UnityEngine.InputSystem.Gamepad: void PauseHaptics(): tag inside `` PVP-150-1: UnityEngine.InputSystem.Gamepad: void ResumeHaptics(): in block context (only allowed in top-level context) PVP-150-1: UnityEngine.InputSystem.Gamepad: void ResumeHaptics(): tag inside ` PVP-150-1: UnityEngine.InputSystem.Gamepad: void ResetHaptics(): in block context (only allowed in top-level context) PVP-150-1: UnityEngine.InputSystem.Gamepad: void ResetHaptics(): tag inside ` ``

The `<inheritdoc />` tag needs to be used at the top-level (it can't be a in-block tag). I fixed that as best as possible.

File: Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs [hunk 10]
```diff
@@ -399,109 +402,96 @@ namespace UnityEngine.InputSystem
...
-    /// Debug.Log(string.Join("\n", Gamepad.all));
-    ///
-    /// // Check whether the X button on the current gamepad is pressed.
-    /// if (Gamepad.current.xButton.wasPressedThisFrame)
-    ///     Debug.Log("Pressed");
-    ///
-    /// // Rumble the left motor on the current gamepad slightly.
-    /// Gamepad.current.SetMotorSpeeds(0.2f, 0.
-    /// </code>
+    /// <code source="../../DocCodeSamples.Tests/GamepadExample.cs" />
```
Comment (ritamerkl) [inline Packages/com.unity.inputsystem/InputSystem/Devices/Gamepad.cs:407]: cool! That is handy!

File: Packages/com.unity.inputsystem/DocCodeSamples.Tests/GamepadExample.cs [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+        // Check if the button control is being continuously actuated and read its value
+        if (gamepad.rightTrigger.IsActuated())
+        {
+            Debug.Log("Right trigger value: " + gamepad.rightTrigger.ReadValue());
+        }
+
+        // Read left stick value and perform some code based on the value
+        Vector2 move = gamepad.leftStick.ReadValue();
+        {
+            // Do 'move' code here
```
Comment (ritamerkl) [inline Packages/com.unity.inputsystem/DocCodeSamples.Tests/GamepadExample.cs]: Sounds a bit odd
// Use the Vector2 move for the game logic

**Review by ritamerkl — COMMENTED** [review]
LGTM, only one minor comment - Despite CI seems to fail, approving to unblock you

**Review by ekcoh — APPROVED** [review]
LGTM, nice to see the referenced doc class under asmdef

**Pull request merged**